api/v1beta1/vaultauth_types.go

```diff
@@ -90,6 +90,31 @@ type VaultAuthConfigAWS struct {
 	IRSAServiceAccount string `json:"irsaServiceAccount,omitempty"`
 }
 
+// VaultAuthConfigGCP provides VaultAuth configuration options needed for
+// authenticating to Vault via a GCP AuthMethod, using workload identity
+type VaultAuthConfigGCP struct {
+	// Vault role to use for authenticating
+	Role string `json:"role"`
+
+	// WorkloadIdentityServiceAccount is the name of a Kubernetes service
+	// account (in the same Kubernetes namespace as this VaultAuth) which has
+	// been configured for workload identity in GKE. Should be annotated with
+	// "iam.gke.io/gcp-service-account".
+	WorkloadIdentityServiceAccount string `json:"workloadIdentityServiceAccount"`
+
+	// GCP Region of the GKE cluster's identity provider. Defaults to the region
+	// returned from the operator pod's local metadata server.
+	Region string `json:"region,omitempty"`
+
+	// GKE cluster name. Defaults to the cluster-name returned from the operator
+	// pod's local metadata server.
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// GCP project id. Defaults to the project-id returned from the operator
+	// pod's local metadata server.
+	ProjectId string `json:"projectId,omitempty"`
```

Comment:
nit: Not sure if this is consistently followed everywhere, but ID is the Go naming convention, and we do use it for RoleID above.

```suggestion
	ProjectID string `json:"projectId,omitempty"`
```